components/gantry/oneaxis/oneaxis.go

```diff
@@ -187,10 +189,79 @@ func (g *oneAxis) Reconfigure(ctx context.Context, deps resource.Dependencies, c
 			return err
 		}
 	}
+	ctx, cancelFunc := context.WithCancel(context.Background())
+	g.cancelFunc = cancelFunc
+	g.checkHit(ctx)
 
 	return nil
 }
 
+func (g *oneAxis) checkHit(ctx context.Context) {
+	g.activeBackgroundWorkers.Add(1)
+	utils.PanicCapturingGo(func() {
+		defer utils.UncheckedErrorFunc(func() error {
+			return g.motor.Stop(ctx, nil)
+		})
+		defer g.activeBackgroundWorkers.Done()
+		for {
+			if ctx.Err() != nil {
+				return
+			}
+			select {
+			case <-ctx.Done():
+				return
+			default:
+			}
+			func() {
+				for i := 0; i < len(g.limitSwitchPins); i++ {
+					hit, err := g.limitHit(ctx, i)
+					if err != nil {
+						g.logger.Error(err)
+						return
+					}
+					if hit {
+						if err := g.moveAway(ctx, i); err != nil {
+							g.logger.Error(err)
+							return
+						}
+					}
+				}
+			}()
+		}
+	})
+}
+
+// Once a limit switch is hit in any move call (from the motor or the gantry component),
+// this function stops the motor, and reverses the direction of movement until the limit
+// switch is no longer activated.
+func (g *oneAxis) moveAway(ctx context.Context, pin int) error {
+	if err := g.motor.Stop(ctx, nil); err != nil {
+		return err
+	}
+	time.Sleep(100 * time.Millisecond)
+	if pin == 0 {
+		if err := g.motor.GoFor(ctx, g.rpm, 0, nil); err != nil {
+			return err
+		}
+	} else {
+		if err := g.motor.GoFor(ctx, -1.0*g.rpm, 0, nil); err != nil {
+			return err
```

Comment:
Lines 247-249 are identical to lines 243-245 except for the -1, yeah? I'd pull out the shared code.
```go
var direction = 1.0
if pin != 0 {
    direction = -1.0
}
if err := g.motor.GoFor(ctx, direction * g.rpm, 0, nil); err != nil {
    return err
}
```
That way, it's clearer what's different and what's the same, and if we ever need to change this part, we only need to change it once.